geomesa-spark/geomesa-spark-converter/src/main/scala/org/locationtech/geomesa/spark/converter/ConverterSpatialRDDProvider.scala

```diff
@@ -84,17 +76,49 @@ class ConverterSpatialRDDProvider extends SpatialRDDProvider with LazyLogging {
     rdd.map(_._2)
   }
 
+  private def computeSftConfig(params: Map[String, String], query: Query) = {
+    if (params.contains(ConverterNameKey)) {
+      // NB: Here we assume that the SFT name and Converter name match.  (And there is no option to rename the SFT.)
+      //  Further, it is assumed that they can loaded from the classpath.
+      val sftName = params.get(ConverterNameKey).orElse(Option(query.getTypeName)).orNull
+      val sft = SimpleFeatureTypeLoader.sftForName(sftName)
+        .getOrElse(throw new IllegalArgumentException(s"Couldn't get look up converter by name for $sftName."))
+      val convertConf = ConverterConfigLoader.confs(sftName).root().render(ConfigRenderOptions.concise())
+
+      (sft, convertConf)
+    } else {
+      // This is the general case where the
+      val sftName = params.get(FeatureNameKey).orElse(Option(query.getTypeName)).orNull
+
+      val sft: SimpleFeatureType = SftArgResolver.getArg(SftArgs(params(SftKey), sftName)) match {
+        case Right(s) => s
+        case Left(e) => throw new IllegalArgumentException("Could not resolve simple feature type", e)
+      }
+
+      val converterConf: String = params(ConverterKey)
+      try {
+        CloseQuietly(SimpleFeatureConverters.build(sft, ConfigFactory.parseString(converterConf)))
+      } catch {
+        case NonFatal(e) => throw new IllegalArgumentException("Could not resolve converter", e)
+      }
+
+      (sft, converterConf)
+    }
+  }
+
   override def save(rdd: RDD[SimpleFeature], writeDataStoreParams: Map[String, String], writeTypeName: String): Unit =
     throw new NotImplementedError("Converter provider is read-only")
 }
 
 object ConverterSpatialRDDProvider {
   // converter definition as typesafe config string
-  val ConverterKey   = "geomesa.converter"
+  val ConverterKey     = "geomesa.converter"
   // simple feature type name (optional)
-  val FeatureNameKey = "geomesa.sft.name"
+  val FeatureNameKey   = "geomesa.sft.name"
   // input file paths, comma-delimited
-  val InputFilesKey  = "geomesa.converter.inputs"
+  val InputFilesKey    = "geomesa.converter.inputs"
   // simple feature type, as spec string, config string, or environment lookup
-  val SftKey         = "geomesa.sft"
+  val SftKey           = "geomesa.sft"
+  // Converter name (to be looked up from the classpath).
+  val ConverterNameKey = "geomesa.converter.name"
```

Comment:
Excellent point.  The tough thing I'm trying to sort out is all the combinations we wish to support.

Eventually, I'd like users to be able to mix n' match looking up sfts or converters with providing explicit config.  Do some follow-on tickets make sense for this and other issues?